client/src/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,93 @@
+import type { CalendarProps, Row } from "@/components/Calendar/calendar.types";
+import { useCalendar } from "@/components/Calendar/hooks/use-calendar";
+import useMonthPicker from "@/components/Calendar/hooks/use-month-picker";
+import { daysOfWeekShort } from "@/lib/datetime/weekdays";
+import type { Maybe } from "@/types/server/utility.types";
+import { MonthPicker } from "@mantine/dates";
+import type { Dayjs } from "dayjs";
+import S from "./Calendar.style";
+
+type CalendarRowProps = {
+	month: number;
+	year: number;
+	row: Row;
+	selectDate: (day: number) => void;
+	selectedDate?: Maybe<Dayjs>;
+};
+
+function CalendarRow({ month, year, row, selectDate, selectedDate }: CalendarRowProps) {
+	function isSelected(day: number | null) {
+		if (!selectedDate || !day) return false;
+		return (
+			day === selectedDate.date() &&
+			month === selectedDate.month() &&
+			year === selectedDate.year()
+		);
+	}
+
+	return (
+		<S.Row>
+			{row.map((day, index) => (
+				<S.Cell
+					disabled={day === null}
+					key={index}
+					onClick={() => (day ? selectDate(day) : undefined)}
```

Comment:
Don't like this syntax. Maybe don't a ternary, or handle case for `null` days inside `selectDate` itself.